coala command: use --limit-files

fleimgruber opened the pull request:
Uses `--limit-files` instead of `--files`.  Also, additional line-start
and line-end regexp patterns.

fleimgruber committed:
coala command: Use --limit-files

Uses `--limit-files` instead of `--files`.  coafile: Ignore for gitmate
CI and include for ert.

sils commented:
Comment on 0fd59448cfad4c41e1ff1a2e6b42a88808c311c3, file test/resources/language/pep8.py, line 1.

The code does not comply to PEP8.

*PEP8Bear, severity NORMAL, section `python`.*

The issue can be fixed by applying the following patch:

```diff
--- a/test/resources/language/pep8.py
+++ b/test/resources/language/pep8.py
@@ -1 +1 @@
-a = 1 + 1 # <- not PEP8 compliant
+a = 1 + 1  # <- not PEP8 compliant
```

fleimgruber commented:
@sils So this is the separate PR mentioned in https://github.com/coala/coala-emacs/pull/17.  This is really how `coala` should be called in this case - I even vaguely remember you telling me just that during the sprint.

The issue now is that the PEP8 test file causes `coala` to complain when checking the repo itself.  Can you somehow force this to be accepted despite that?

sils commented on .coafile:
```diff
@@ -6,3 +6,4 @@ bears = GitCommitBear
 
 [python]
 bears = PEP8Bear
+files = **.py
```
as per gitmate result https://github.com/coala/coala-emacs/pull/18#issuecomment-317204432 maybe put the test files into an ignore?

fleimgruber replied:
see https://github.com/coala/coala-emacs/pull/18/commits/875adbc73d53a0f8323413d5117caec91c5c51a2 which ignores the test file from root and includes it in the test subfolder (so that `cask exec ert-runner` picks it up).  Test works, a local `coala` run from the root ignores the test file, but gitmate is still complaining.  What am I missing?  Is there a separate gitmate config?

9r0x commented:
Anyone working on this repo now? I guess 'Default' is deprecated now.

jayvdb commented:
ping @fleimgruber

fleimgruber commented:
Not currently working on it.  Will have a look in the next days.

fleimgruber commented:
@Grox-Ni the last commit already contains that deprecation.

fleimgruber commented:
FYI, I am using this on a daily basis and just tried with the most recent PyPI dev version and that works as well.  AFAICS, the remaining problem is that gitmate-bot does not honour the "ignore" directive, see also https://github.com/coala/coala-emacs/pull/18#discussion_r129094580.

9r0x commented:
@fleimgruber you are right, I have an idea to solve gitmate's problem, please see #20, do you think I should reopen it?

fleimgruber commented:
@Grox-Ni No.  What @jayvdb said in https://github.com/coala/coala-emacs/pull/20#issuecomment-342408430 and following.

nemani commented:
@fleimgruber Are you still on this? Or should I close this? :)

fleimgruber commented:
@nemaniarjun this is still waiting for feedback on CI failures because of gitmate-bot, see https://github.com/coala/coala-emacs/pull/18#discussion_r129094580.

nemani commented:
@fleimgruber @Makman2 I am confused here, which config does gitmate bot use, the one in the PR or the one on master? Also can't we just override the CI so that we can get this merged? (I mean if gitmate uses the coafile from master then we should just merge this)

Makman2 commented:
I believe it takes it from the PR, otherwise you can't fix issues once they are on master.

Makman2 commented:
However I don't see a problem adding the test files to the `ignore` setting into the coafile if I see it correctly, should be quite straightforward :+1:

Makman2 commented:
Ah sorry haven't read it properly. Let's try to rebase and see if it works. If not, we have to investigate.

Makman2 commented:
@gitmate-bot rebase

Makman2 commented:
Once more, gitmate was disabled.

Makman2 commented:
@gitmate-bot rebase

Makman2 commented:
Hm seems the commits are already up-to-date, could you regenerate the commit hash @fleimgruber and repush? --> `git commit --amend --no-edit && git push -f`

fleimgruber commented:
Done.  gitmate still complains.

Makman2 commented:
Sorry for responding that late, I've become a bit busy and haven't worked through my notification queue^^

And I know the cause: I've actually missed your very first commit in this PR, because it was at the very top of this page. You have to squash the commits, gitmate will do its analysis on each commit and uses the repository state of it. So your commit that added the new filenames in .coafile does not ignore the test file yet and flags a problem.

fleimgruber commented:
@Makman2 well, how obvious it is in hindsight... done and gitmate does not complain anymore.

Makman2 commented on flycheck-coala.el:
```diff
@@ -34,10 +34,10 @@
-  :error-patterns  ((error (or "None" line) ":" (or "None" column) ":MAJOR:" (message))
-                    (warning (or "None" line) ":" (or "None" column) ":NORMAL:" (message))
-                    (info (or "None" line) ":" (or "None" column) ":INFO:" (message)))
+  :command ("coala" "--format" "{line}:{column}:{severity_str}:{message}" "--find-config" "--limit-files" source-original)
```
But won't `limit-files` filter *out* given files? Wouldn't this render the plugin completely useless?

fleimgruber replied:
https://github.com/coala/coala-emacs/commit/04adb144cda16535359f3be9fa492eb52c0e7120 changed `--limit-files` to `--files`.  The `--limit-files` was put in https://github.com/coala/coala-emacs/commit/88315e627b4ff8f04629a312e4b453d939a65bf9. Also, appeal to authority: During the hackathon, this was suggested to me by @sils.

Makman2 commented on flycheck-coala.el:
```diff
@@ -34,10 +34,10 @@
+  :command ("coala" "--format" "{line}:{column}:{severity_str}:{message}" "--find-config" "--limit-files" source-original)
+  :error-patterns  ((error line-start (or "None" line) ":" (or "None" column) ":MAJOR:" (message) line-end)
+                    (warning line-start (or "None" line) ":" (or "None" column) ":NORMAL:" (message) line-end)
+                    (info line-start (or "None" line) ":" (or "None" column) ":INFO:" (message) line-end))
```
Don't you have to adapt the coala `--format` to also emit line-starts and line-ends?

fleimgruber replied:
See `:error-patterns` documentation and example in http://www.flycheck.org/en/latest/developer/developing.html#writing-the-checker. The `line-start` and `line-end` are `rx` pattern elements and do not map to checker command results.

Makman2 commented:
1. Commit message:
   - ~~C~~coala :D
   - ~~u~~Use
2. I see two commits here:
   > Uses `--limit-files` instead of `--files`.  Also, additional line-start and line-end regexp patterns.

   One change for `--limit-files` and one for the additional patterns.

fleimgruber commented:
Thanks for review. I addressed your comments, changed the commit message (typos) and reverted the regexp pattern change. With your consent I would squash.

Makman2 commented:
Alright you can squash ;) Please rectify your final commit message so things like `line-end` aren't mentioned any more (due to your "revert commit" :) )

fleimgruber commented:
@Makman2 Squashed.
